File: 3447-clear-digits/3447-clear-digits.java
class Solution {
    public String clearDigits(String s) {
        int i = 0;
        while (s.matches(".*\\d.*")) {
            if (Character.isDigit(s.charAt(i))) {
                s = s.substring(0, i - 1) + s.substring(i + 1);
                i = 0;
            } else {
                i++;
            }
        }

        return s;
    }
}

File: 3447-clear-digits/3447-clear-digits.cpp
class Solution {
public:
    string clearDigits(string s) {
        string result = "";

        for (int i = 0; i < s.length(); i++) {
            if (isdigit(s[i])) {
                result.erase(result.length() - 1);
            } else {
                result += s[i];
            }
        }

        return result;
    }
};